src/linearmodel.jl

```diff
@@ -187,41 +212,275 @@ function discrete_jacobian!(::Type{DiscreteSystemQuadrature}, ∇f, model::Discr
+        RobotDynamics.get_times(model::($name)) = model.times
+        RobotDynamics.set_A!(model::($name), A::AbstractArray, k::Integer) = model.A[k] = A
+        RobotDynamics.set_B!(model::($name), B::AbstractArray, k::Integer) = model.B[k] = B
+        RobotDynamics.set_d!(model::($name), d::AbstractArray, k::Integer) = model.d[k] = d
+        RobotDynamics.set_times!(model::($name), times::AbstractVector) = model.times .= times
+
+        function ($name)()
+            A_vec = [@SMatrix zeros($n, $n) for _ = 1:($N-1)]
+            B_vec = [@SMatrix zeros($n, $m) for _ = 1:($N-1)]
+            d_vec = [@SVector zeros($n) for _ = 1:($N-1)]
+            times = zeros($N)
+
+            TA = eltype(A_vec)
+            TB = eltype(B_vec)
+            Td = eltype(d_vec)
+            T = eltype(times)
+
+
+            return ($name){TA, TB, Td, T}(A_vec, B_vec, d_vec, times)
+        end
+    end
+
+    esc(Expr(:toplevel,
+            struct_exp,
+            function_def))
 end
 
-function _discretize(::Type{Exponential}, ::Val{true}, model::ContinuousLinearModel, k::Integer, dt)
-    A_c = get_A(model, k)
-    B_c = get_B(model, k)
-    n = size(A_c, 1)
-    D_c = oneunit(SizedMatrix{n, n})
-    m = size(B_c, 2)
+"""
+    @create_discrete_lti(name, n, m, is_affine=false)
+
+This macro can be used to conveniently create a DiscreteLTI model subtype and model instance.
+In order to use the resulting model, use the `set_A!`, `set_B!`, and `set_d!` functions to 
+add in your system matrices. The macro defines a default constructor with system matrices of 
+all zeros. This macro must be called from a top level scope.
+
+Creates a struct that looks like: 
+```julia
+struct (name){TA, TB, Td} <: DiscreteLTI
+    A::Base.RefValue{TA}
+    B::Base.RefValue{TB}
+    d::Base.RefValue{Td}
+end
+```
+All functions necessary to modify/use the struct are predefined by the macro. The macro defines a 
+default constructor using `SMatrix` with all zeros. This can be called simply with `(name)()`.
+If `is_affine = false` then the d term is not defined in the struct.
+"""
+macro create_discrete_lti(name, n, m, is_affine=false)
+    if is_affine   
+        _lti_affine(name, n, m, :DiscreteLTI)
+    else
+        _lti_non_affine(name, n, m, :DiscreteLTI)
+    end
+end
 
-    continuous_system = zero(SizedMatrix{(2*n)+m, (2*n)+m})
-    continuous_system[1:n, 1:n] .= A_c
-    continuous_system[1:n, n .+ (1:m)] .= B_c
-    continuous_system[1:n, n + m .+ (1:n)] .= D_c
+"""
+    @create_continuous_lti(name, n, m, is_affine=false)
+
+This macro can be used to conveniently create a ContinuousLTI model subtype and model instance.
+In order to use the resulting model, use the `set_A!`, `set_B!`, and `set_d!` functions to 
+add in your system matrices. The macro defines a default constructor with system matrices of 
+all zeros. This macro must be called from a top level scope.
+
+Creates a struct that looks like: 
+```julia
+struct (name){TA, TB, Td} <: ContinuousLTI
```

Comment:
Are you using `RefValue` to allow the values to change? I'm curious if this is any more efficient than just making the struct mutable? I think I'd prefer to make the struct mutable since it'll probably make it easier to understand what's going on.